src/core/MRApp.js

```diff
@@ -202,6 +202,16 @@ export class MRApp extends MRElement {
         this.renderer.toneMapping = THREE.ACESFilmicToneMapping;
         this.renderer.toneMappingExposure = 1;
         this.renderer.localClippingEnabled = true;
+        if (this.getAttribute('preserve-drawing-buffer') ?? false) {
+            // There's issues in the timing to enable taking screenshots of threejs scenes unless you have direct access to the code.
+            // Using the preserveDrawingBuffer to ignore timing issues is the best approach instead. Though this has a performance hit,
+            // we're allowing it to be enabled by users when necessary.
+            //
+            // References:
+            // https://stackoverflow.com/questions/15558418/how-do-you-save-an-image-from-a-three-js-canvas
+            // https://stackoverflow.com/questions/30628064/how-to-toggle-preservedrawingbuffer-in-three-js
+            this.renderer.preserveDrawingBuffer = true;
```

Comment:
`preserveDrawingBuffer` flag must be passed via the `WebGLRenderer` constructor.

https://threejs.org/docs/#api/en/renderers/WebGLRenderer

Reply:
good catch with that

---

very odd, i feel like im not seeing any difference using it in constructor, outside, or even not having it at all

i wonder if it's a browser thing - i'll look into this and follow back up for why the discrepancy as we might not even need the flag then if it works?

weird...

Reply:
discussed - following up per [conversation](https://github.com/Volumetrics-io/mrjs/issues/506#issuecomment-2007921050) for why issue wasnt showing up on my end - and [added note about chromium base](https://github.com/Volumetrics-io/documentation/pull/26/commits/9f623225d10f8ad7e0d99e0e815032ab8bca2dde) to the documentation pr

Reply:
going to continue to implement a working fix ~